### PR TITLE
Add FTM and PWT driver pwm support for NXP frdm_ke17z and frdm_ke17z512

### PR DESCRIPTION
**Pull request by ankeXiao**
Tested `samples/basic/blinky_pwm`, `samples/basic/fade_led`, `tests/drivers/pwm/pwm_api` and `tests/drivers/pwm/pwm_loopback` samples.

### CODE REVIEW

File: tests/drivers/pwm/pwm_loopback/src/test_pwm_loopback.c [hunk 1]
```diff
@@ -125,6 +126,11 @@ ZTEST_USER(pwm_loopback, test_pulse_capture_inverted)
 
 ZTEST_USER(pwm_loopback, test_period_capture)
 {
+#if defined(CONFIG_SOC_SERIES_KE1XZ)
+	/* Skipping normal-polarity period capture in PWT tests isn't supported by ke1xz */
+	TC_PRINT("skip the period capture test in normal polarity\n");
```
Comment (decsny) [inline tests/drivers/pwm/pwm_loopback/src/test_pwm_loopback.c:131]: this should be handled by the driver, not by adding platform specific code to the test. You will notice that test_capture() will already skip if the driver returns -ENOTSUP when requesting this configuration.

**Review by decsny — CHANGES_REQUESTED** [review]

File: tests/drivers/pwm/pwm_loopback/src/test_pwm_loopback.c (hunk 1 above)
Comment (ankeXiao) [reply tests/drivers/pwm/pwm_loopback/src/test_pwm_loopback.c]: Thanks for reminding, I took this code to the driver, for not supported will return -ENOTSUP

File: drivers/pwm/pwm_mcux_pwt.c [hunk 2]
```diff
@@ -85,6 +86,20 @@ static int mcux_pwt_configure_capture(const struct device *dev,
...
 	}
 
+#if defined(CONFIG_SOC_SERIES_KE1XZ)
+	if ((flags & PWM_CAPTURE_TYPE_MASK) == PWM_CAPTURE_TYPE_BOTH) {
+		LOG_ERR("Cannot capture both period and pulse width");
+		return -ENOTSUP;
+	}
+
+	if ((flags & PWM_CAPTURE_TYPE_MASK) == PWM_CAPTURE_TYPE_PERIOD) {
+		if ((flags & PWM_POLARITY_MASK) == PWM_POLARITY_NORMAL) {
```
Comment (decsny) [inline drivers/pwm/pwm_mcux_pwt.c:96]: nit: instead of nesting conditionals, just use an &&

Comment (ankeXiao) [reply drivers/pwm/pwm_mcux_pwt.c:96]: Thanks for reminding, I will update it.

**Review by decsny — DISMISSED** [review]

**Review by EmilioCBen — DISMISSED** [review]

**Comment by decsny** [comment]
@ankeXiao you need to fix the merge conflicts

**Review by decsny — DISMISSED** [review]

**Comment by ankeXiao** [comment]
Fixed all merge conflicts

**Comment by ankeXiao** [comment]
Hi all, please help approve and merge this PR, thanks!

**Review by decsny — APPROVED** [review]

**Comment by ankeXiao** [comment]
@anangl, could you provide your review so it can move forwards?

**Review by anangl — APPROVED** [review]

**Pull request merged**